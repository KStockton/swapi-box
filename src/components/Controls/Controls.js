import React, { Component } from 'react';

class Controls extends Component {



  // handleClick = (event) =>{
  //   const { name } = event.target
  //    console.log(event.target.name)
  //   this.setState({})
  // }
  
  
  render() {
    console.log(this.state)
  return(
    <section>
      <h1>Swapi-Box</h1>
      <button name="people" onClick={this.handleClick}>People</button>
      <button name="planets" onClick={this.handleClick}>Planets</button>
      <button name="vehicles" onClick={this.handleClick}>Vehicles</button>
    </section>
  )
  }
  }

export default Controls