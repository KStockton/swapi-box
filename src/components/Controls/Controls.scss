.Controls-btn {
  margin-left: 95px;
  width: 163px;
  height: 58px;
  border-radius: 8px;
  box-shadow: inset 4px 3px 15px #b8b8bd;
  font-size: 21px;
  font-family: Orbitron;
  padding-top: 3px;
  font-weight: 700;
  cursor: pointer;
}

.people:hover {
  background-color: #ffff00;
}

.planets:hover {
  background-color: #ffff00;
}

.vehicles:hover {
  background-color: #ffff00;
}

.vehicles-wrapper {
  display: flex;
  justify-content: center;
  align-items: center;
}

@media only screen and (max-width: 800px) {
  .title {
    width: 22rem;
  }
  .Controls-wrapper {
    display: flex;
    flex-direction: column;
    justify-content: center;
    margin: 0px auto;
  }
  .Controls-btn {
    display: flex;
    height: 3rem;
    width: 185px;
    margin-bottom: 27px;
  }

  .filmtext-wrapper-after {
    position: absolute;
    top: 100%;
    right: 27%;
    animation: scrollCrawl 50s linear;
    animation-iteration-count: 1;
    color: #ffd900;
    letter-spacing: -2px;
    width: 11rem;
    font-size: 106%;
    text-align: justify;
    text-transform: uppercase;
    font-family: Pathway Gothic One sans-serif;
    background: linear-gradient(0deg, transparent, #0000001a 75%);
    line-height: 105%;
  }

  .filmtext-wrapper-before {
    position: fixed;
    left: 5%;
    bottom: 0%;
    overflow: hidden;
    transform: perspective(250px) rotateX(41deg);
    height: 80%;
    width: 90%;
}
.Favorites-btn {
  display: flex;
  flex-direction: row;
  align-items: center;
  border: 2px solid white;
  width: 172px;
  padding-left: 15px;
  border-radius: 12px;
  margin: 26px;
  font-size: 17px;
  position: absolute;
  text-align: center;
  right: 3%;
  height: 41px;
}
}