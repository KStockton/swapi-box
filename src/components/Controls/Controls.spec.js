import React from 'react';
import Controls from './Controls';
import { shallow } from 'enzyme';

const mock_handleCategory = jest.fn();

describe('Controls', () =>{
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Controls handleCategory={mock_handleCategory}/>);
  });

  it('Expect component to render without crashing', () =>{
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleCategory when any a button is clicked', () => {
    wrapper.find('.Controls-wrapper').childAt(1).simulate('click',  {target: { value: "people"}});
    expect(mock_handleCategory).toBeCalled();
  });
});