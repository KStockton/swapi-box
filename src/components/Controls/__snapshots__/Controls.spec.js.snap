// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls Expect component to render without crashing 1`] = `
<section
  className="Controls-wrapper"
>
  <h1
    className="title"
  >
    Swapi-Box
  </h1>
  <button
    className="Controls-btn people"
    name="people"
    onClick={[Function]}
  >
    <div
      className="vehicles-wrapper"
    >
      <img
        alt="star-trooper"
        src="Trooper.svg"
      />
      People
    </div>
  </button>
  <button
    className="Controls-btn planets"
    name="planets"
    onClick={[Function]}
  >
    <div
      className="vehicles-wrapper"
    >
      <img
        alt="deathstar"
        src="DeathStar.svg"
      />
      Planets
    </div>
  </button>
  <button
    className="Controls-btn vehicles"
    name="vehicles"
    onClick={[Function]}
  >
    <div
      className="vehicles-wrapper"
    >
      <img
        alt="falcon"
        src="Falcon.svg"
      />
      Vehicles
    </div>
  </button>
</section>
`;
